beetsplug/convert.py

```diff
@@ -183,11 +183,11 @@ def encode(self, command, source, dest, pretend=False):
             self._log.info(u'Encoding {0}', util.displayable_path(source))
 
         # Substitute $source and $dest in the argument list.
-        args = shlex.split(command)
+        args = shlex.split(command.decode(ui._arg_encoding()))
         for i, arg in enumerate(args):
             args[i] = Template(arg).safe_substitute({
-                'source': source,
-                'dest': dest,
+                'source': source.decode(ui._arg_encoding()),
+                'dest': dest.decode(ui._arg_encoding()),
             })
```

Comment:
this entire commit is just one way I've attempted to solve the problem of course, and I'm not wedded to it in any way shape or form. That's why  I wondered about us attempting to try using BytesIO and trying to see if maybe we could replace the stdin/stdout/stderr with bytes equivalents. I'm having a really hard time trying to find information on this in a way that works in both windows and linux. I read  a bunch of posts on this, but it's all either outdated or doesn't take into account both linux and windows. It also doesn't help us with stuff like shlex.split anyways